app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.kt

```diff
@@ -123,6 +134,48 @@ interface CommentsViewModel {
                         commentsList.onNext(it.comments)
                     }
                 }
+
+            this.currentUser.loggedInUser()
+                .compose(Transformers.takePairWhen(this.postComment))
+                .compose(bindToLifecycle())
+                .subscribe {
+                    val comment = Comment.builder()
+                        .body(it.second.first)
+                        .parentId(-1)
+                        .authorBadges(listOf())
+                        .createdAt(it.second.second)
+                        .cursor("")
+                        .deleted(false)
+                        .id(-1)
+                        .repliesCount(0)
+                        .author(it.first)
+                        .build()
+
+                    this.insertComment.onNext(comment)
+                }
+
+            initialProject
+                .compose(Transformers.takePairWhen(this.postComment))
+                .compose(bindToLifecycle())
+                .switchMap {
+                    it.first?.let { project ->
+                        this.apolloClient.createComment(
+                            PostCommentData(
+                                project = project,
+                                body = it.second.first,
+                                clientMutationId = null,
+                                parentId = null
+                            )
+                        )
+                    }
+                }
+                .subscribe(
+                    {
+                        this.commentPosted.onNext(it)
+                    },
+                    {
```

Comment:
I guess we do not have defined yet the behaviour in case it fails right?

Reply:
> I guess we do not have defined yet the behaviour in case it fails right?

This behavior is on a separate ticket. I just created the required codes to implement the response.

Reply:
I have added the error observers both in the viewmodel and in the activity